src/main/kotlin/org/rust/ide/intentions/ExpandStructFieldsIntention.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.ide.utils.expandStructFields
+import org.rust.ide.utils.fillRsPatWithFields
+import org.rust.ide.utils.getTupleStructMissingFields
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ref.deepResolve
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+class ExpandStructFieldsIntention : RsElementBaseIntentionAction<ExpandStructFieldsIntention.Context>() {
+    override fun getText() = "Expand actual struct fields"
+
+    override fun getFamilyName() = text
+
+    data class Context(
+        val structBody: RsPat
+    )
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        return if (element.elementType == RsElementTypes.DOTDOT
+            && (element.context is RsPatStruct || element.context is RsPatTupleStruct)) {
+            Context(element.context as RsPat)
+        } else {
+            null
+        }
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val factory = RsPsiFactory(project)
+        val structBody = ctx.structBody
+        if (structBody is RsPatStruct) {
+            val declaration = structBody.path.reference.deepResolve() as? RsFieldsOwner ?: return
+            val bodyFieldNames = structBody.patFieldList.map { it.kind.fieldName }.toSet()
+            structBody.dotdot?.delete()
+            val missingFields = declaration.fields
+                .filter { it.name !in bodyFieldNames && !it.queryAttributes.hasCfgAttr() }
+                .map { factory.createPatField(it.name!!) }
+            val fieldPositions = HashMap<String, Int>(missingFields.size)
+            var position = 0
+            declaration.fields.forEach { fieldPositions[it.name!!] = position++ }
+            expandStructFields(RsPsiFactory(project), structBody, fieldPositions, missingFields)
+        } else if (structBody is RsPatTupleStruct) {
+            val declaration = structBody.path.reference.deepResolve() as? RsFieldsOwner ?: return
+            val bodyFields = structBody.childrenOfType<RsPatIdent>()
+            structBody.dotdot?.delete()
+            val missingFields = getTupleStructMissingFields(factory, declaration.fields.size - bodyFields.size)
+            fillRsPatWithFields(factory, structBody, missingFields)
+            editor.buildAndRunTemplate(ctx.structBody, ctx.structBody.childrenOfType<RsPatBinding>().map { it.createSmartPointer() })
```

Comment:
Let's extract these 2 lines to a separate function called `expandTupleStructFields` and put it to `StrunctFieldsExpander.kt`. This is more convenient with "struct" branch, allows to reduce code duplication and also makes available to make `fillRsPatWithFields` private.